language/move-vm/runtime/src/runtime.rs

```diff
@@ -254,13 +254,15 @@ impl VMRuntime {
         )
     }
 
-    // See Session::execute_function for what contracts to follow.
+    // See Session::execute_function and Session::execute_script_function for what contracts to
+    // follow in executing system and script functions.
     pub(crate) fn execute_function(
         &self,
         module: &ModuleId,
         function_name: &IdentStr,
         ty_args: Vec<TypeTag>,
         args: Vec<Vec<u8>>,
+        senders_opt: Option<Vec<AccountAddress>>,
```

Comment:
I feel like having both `execute_function` and `execute_script_function` implemented in the same path is making this interface unnecessarily complicated. I would suggest leaving `execute_function` unchanged and copy some of the common logic to `execute_script_function`. (It's just a few more lines of "duplicate code" and you won't have to deal with the runtime switch here.)

We should again consider whether we want signer propagation to be a Move VM feature in the long term. Personally, I feel like this behavior is Diem-specific and should therefore be handled by the adapter, and I can see third party clients (e.g. other blockchains) wanting a more general interface. (Although Sam made a good point last time that there has to be some ways for the signers to be instantiated and our current rule is not something unreasonable.)

cc @sblackshear

Reply:
Let's move this part of the discussion to #7584, as that will likely be what we are doing here. Bit of duplicated work on this PR :)

Reply:
The implementation in #7584 addressed @vgao1996's comment, I'll do a rebase as soon as #7584 lands.

Reply:
PR rebased